### ISSUE TITLE
Beta Limiting issue list

### ISSUE THREAD

**Issue by lastmeta** [moontreeapp/raven_front#6]
Here are the final issues that should be resolved before beta release: 

- [x] 0. implement a debug crash report system for beta testers.
- [x] 1. Fix Histories. Incoming UTXOs are fine, but outgoing transaction records fail to be recorded (this should solve 3 obscure bugs): 
  - [x] a. upon reloading after transaction existing histories don't get replaced (updateExistingAddresses)
  - [x] b. upon sending the address sent from doesn't update (still thinks has balance)
  - [x] c. tx list in UI missing outgoing transactions
  - [x] d. tx list initially incomplete, corrects itself on reload. fix.
  - [x] e. fix view of tx
- [x] 2. Client disconnect issue (5 minutes if you haven't used the client you get disconnected, a new connection is established, but something is using the old connection object and gives error)
```
still seeing this on occasion
E/flutter (17518): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: Bad state: The client closed with pending request "blockchain.scripthash.get_transaction".
```
- [x] 3. can you recreate freezing on send issue (if so is it an infinite loop on calculating fee?)
- [x] 4. Implement save json file on android correctly
- [x] 5. Implement import other formats (WIF and private key)
- [x] 6. speed things up (tx generation ~ 3 seconds, import mnemonic ~ 10 (waiting for listeners triggered after wallet generation?)) 
- [x] 6b. fix concurrent modification issue
- [x] 6a. fix the "already Listening" warning
- [x] 6c. please wait on account creation
- [x] 6d. import file for import needs to work
--- nice to have ---
- [ ] 7. fix ravencoin lib so we can add memo on blockchain...
- [ ] 7a. fix ravencoin lib so we can send assets...
- [ ] 8. improve in app spinners 
- [x] 9. improve loading page spinner
- [x] 10. indicator light if stuff it happening in the background?
- [x] 11. client connected indicator light?

- [x] fix keyboard stays up on receive screen
- [x] fix mempool (is not) in transactions
- [x] hide bottom bar until trading platform is built out
- [x] hide get raven
- [x] if the app is in the background and hasn't been used for 5 minutes it should pause establishing a new connection to electrum until the user opens it or navigates to it again. (https://stackoverflow.com/questions/51835039/how-do-i-check-if-the-flutter-application-is-in-the-foreground-or-not)
- [x] verify our usage of electrum never over consumes (server.our_stats)
- [x] we need a mempool service - we don't get notified when something that was in the mempool becomes a real transaction, so we're getting the latest block header we need to attach a trigger to that waiter that says, 'if you have any mempool transactions - go download them again and update the records'
- [x] fix bug with balances calculating wrong
- [x] fix focus on keyboard after login
- [x] wallet screen - show which addresses have histories bold.
- [x] see if asset see balance functionality is correct.

--- refactor unto simplicity ---
- [x] combine waiters that derive leader wallets into 1 function with 2 streams
- [x] produce trigger for when we're done download all tx and done deriving addresses.
- [x] use trigger to await saveDanglingTransactions(client); await services.balance.recalculateAllBalances();
- [ ] refactor to avoid empty state SetState calls

**Comment by lastmeta** [comment]
- [ ] rounded send receive buttons?
- [ ] scan qr code error
- [ ] scan qr code font size
- [ ] hide send receive on scroll down, show on scroll up
- [ ] "all transactions"
- [ ] show moon tree logo at top instead of "wallet"
